cgi/user2.pl

```diff
@@ -0,0 +1,361 @@
+			}
+		}
+	}
+
+	if ($type eq 'edit_owner') {
+		ProductOpener::Users::check_edit_owner($user_ref, \@errors);
+	}
+	elsif ($type ne 'delete') {
+		ProductOpener::Users::check_user_form($type, $user_ref, \@errors);
+	}
+
+	if ($#errors >= 0) {
+		if ($type eq 'edit_owner') {
+			$action = 'none';
+		}
+		else {
+			$action = 'display';
+		}
+	}
+}
+
+$template_data_ref->{action} = $action;
+$template_data_ref->{errors} = \@errors;
+
+$log->debug("user form - before display / process", { type => $type, action => $action, userid => $userid }) if $log->is_debug();
+
+if ($action eq 'display') {
+
+    # We can pre-fill the form to create an account using the username and password
+	# passed in a form to open a session.
+	# e.g. when a non-logged user clicks on the "Edit product" button
+
+	if (($type eq "add") and (defined param("user_id"))) {
+		my $user_info = remove_tags_and_quote(param('user_id'));
+		$user_info =~ /^(.+?)@/;
+		if ( defined ($1) ){
+			$user_ref->{email} = $user_info;
+			$user_ref->{userid} = $1;
+			$user_ref->{name} = $1;
+			$user_ref->{password} = $new_user_password;
+		}
+		else{
+			$user_ref->{userid} = $user_info;
+			$user_ref->{name} = $user_info;
+			$user_ref->{password} = $new_user_password;
+		}
+	}
+
+	$template_data_ref->{user_ref} = $user_ref;
+	$template_data_ref->{user_id_field} = $user_ref->{userid};
+	$template_data_ref->{user_password_field} = $user_ref->{password};
+
+	# Create the list of sections and fields
+
+	$template_data_ref->{sections} = [];
+
+	if ($user_ref) {
+		push @{$template_data_ref->{sections}}, {
+			id => "user",
+			fields => [
+				{
+					field => "name"
+				},
+				{
+					field => "email",
+					type => "email",
+				},
+				{
+					field => "userid",
+					label2 => "username"
+				},
+				{
+					field => "password",
+					type => "password",
+					label2 => "password"
+				},
+				{
+					field => "confirm_password",
+					type => "password",
+					label2 => "password_confirm"
+				},
+			]
+		};
+
+		# Professional account
+		push @{$template_data_ref->{sections}}, {
+			id => "professional",
+			fields => [
+				{
+					field => "pro",
+					type => "checkbox",
+				},
+				{
+					field => "pro_checkbox",
+					type => "hidden",
+					value => 1,
+				},
+				{
+					field => "requested_org",
+				}
+			]
+		};
+
+		my $team_section_ref = { id => "teams", fields => [] };
+		for (my $i = 1; $i <= 3; $i++) {
+			push @{$team_section_ref->{fields}}, {
+				field => "team_". $i,
+				label2 => sprintf(lang("team_s"), $i),
+			 };
+		};
+
+		push @{$template_data_ref->{sections}}, {%$team_section_ref};
+
+		my $administrator_section_ref = { id => "administrator", fields => [] };
+		push  @{$administrator_section_ref->{fields}}, {
+			field => "org",
+		};
+		foreach my $group (@user_groups) {
+			push @{$administrator_section_ref->{fields}}, {
+				field =>  "user_group_". $group,
+				label2 =>  "user_group_" . ${group} . "_description",
```

Comment:
in the org template, there is a "description" that could be used for what you put in "label2"

e.g. in org.pl it's populated automatically if we have a "[fieldid]_description" entry in the .po files:

			# Label
			$field_ref->{label} = lang($field_lang_id);
			
			# Descriptions and notes for fields
			if (lang($field_lang_id . "_description")) {
				$field_ref->{description} = lang($field_lang_id . "_description");
			}
			if (lang($field_lang_id . "_note")) {
				$field_ref->{note} = lang($field_lang_id . "_note");
			}

Reply:
yes you are right, we can use the description here instead of `label2` in the administrator section.